src/main/java/org/jabref/gui/libraryproperties/saving/SavingPropertiesViewModel.java

```diff
@@ -65,11 +62,9 @@ public void setValues() {
             case TABLE -> saveInTableOrderProperty.setValue(true);
         }
 
-        List<Field> fieldNames = new ArrayList<>(FieldFactory.getCommonFields());
-        fieldNames.add(InternalField.TYPE_HEADER); // allow entrytype field as sort criterion
-        fieldNames.sort(Comparator.comparing(Field::getDisplayName));
+
         sortableFieldsProperty.clear();
-        sortableFieldsProperty.addAll(fieldNames);
+        sortableFieldsProperty.addAll(FieldFactory.getAllFieldsSortedWithoutInternalExceptSome());
```

Comment:
Can we please not blow up the FieldFactory with special methods for exactly one place where the method is used?
Why not just call getAllFields in and add InternalFields stuff here?
Single responsibility!